examples/MicroserviceExample/WorkerService/Worker.cs

```diff
@@ -0,0 +1,68 @@
+﻿// <copyright file="Worker.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using OpenTelemetry;
+using OpenTelemetry.Trace;
+using Utils.Messaging;
+
+namespace WorkerService
+{
+    public partial class Worker : BackgroundService
+    {
+        private readonly MessageReceiver messageReceiver;
+        private readonly TracerProvider tracerProvider;
+
+        public Worker(MessageReceiver messageReceiver)
+        {
+            this.messageReceiver = messageReceiver;
+
+            this.tracerProvider = Sdk.CreateTracerProvider((builder) =>
```

Comment:
..lets come back to this. If program.cs sets up everything, then Worker can simply retrieve TracerProvider from DI in constructor.

Reply:
Ok. Yea, I may have been doing something wrong here. I'll follow up and clean this up.

Reply:
Ok, this is fixed now.